pkg/datastore/test/datastore.go

```diff
@@ -39,9 +39,36 @@ type TestableDatastore interface {
 	ExampleRetryableError() error
 }
 
-// AllExceptWatch runs all generic datastore tests on a DatastoreTester, except
-// those invoking the watch API.
-func AllExceptWatch(t *testing.T, tester DatastoreTester) {
+type Categories map[string]struct{}
```

Comment:
not worth changing everything just for this, but we might want to consider ginkgo for these tests. There are a lot of built-in features that would simplify this for us (i.e. there's an existing `Label` system for labelling / running tests with labels.